adoptingerlang.org

```diff
@@ -3444,44 +3446,74 @@ initContainers:
-
-The only necessary thing to do for graceful shutdown is to delay =init:stop()= from running until the =Pod= is removed from the =kube-proxy= =Endpoint=. The process can wait up to 30 seconds before Kubernetes sends a =SIGKILL= signal that will force the process to terminate.
-
-In =service_discovery= the =readinessProbe= is on a separate path, =/ready=, and with a higher frequency, every 5 seconds.
+**** Readiness, Liveness and Startup Probes
+
+Each container in a =Pod= can define a =readiness=, =liveness= and =startup= probe. The
+=liveness= probe . A startup probe can be added to delay when the =liveness= and
+=readiness= probes start being checked.
+
+Probes are defined either as a command to execute in the container, as an HTTP
+=GET= request made to a given port and path, or as an attempt to open a TCP
+connection to a specific port.
+
+A =livenessProbe= is not required to ensure restart of failed containers. If the
+process in the container crashes or stops for any reason, it will already be
+restarted. The probe should only be used for cases the process does not crash
+but for some reason has deadlocked and can not respond to requests. For this
+reason if a =livenessProbe= is included it should be kept extremely simple. It is
+more often than not safer to leave out the =livenessProbe= entirely and rely on
+the node crashes, metric based alerting and auto-scaling.
+
+One part of auto-scaling is based on the =readinessProbe=. The =readinessProbe=
+tells Kubernetes the =Pod= is ready to receive traffic. When a =Pod= becomes =unready=
+because a container is failing the =readinessProbe=, it is removed from the
+=Service= resulting in traffic being directed at the existing =Pods=, increasing
+their load, then a new =Pod= will be started.
+
+Not using the =livenessProbe= for this purpose means the containers are still
+available, since they aren't killed, and the reason for failing the check can be
+examined. This may mean attaching a shell if it isn't completely frozen or
+forcing a crash dump to be written. In a very resource constrained environment
+you may prefer to have the =livenessProbe= in order to have deadlocked containers
+and their =Pods= cleaned up right away, instead of waiting for manual intervention
+and the auto-scaler to scale down after load falls.
+
+Another use for the =readinessProbe= is if the application has to periodically do work
+where you don't want to be handling requests, or have to do some sort of
+maintenance, it can return a =503= status response for readiness causing it to be
+removed from the =Service= backends.
```

Comment:
Specifically here a common use is to allow draining connections while doing remaining work before shutting down.

Reply:
Na, it isn't used for that. During shutdown the Pod is removed from the Service already. I don't know that it'd hurt, but it isn't necessary.